testsuite/model/src/test/java/org/keycloak/testsuite/model/user/UserSyncTest.java

```diff
@@ -270,17 +271,15 @@ public void testInvalidUsersAreDeleted() {
         });
 
         // import user
-        withRealm(realmId, (session, realm) -> {
+        String oldUserId = withRealm(realmId, (session, realm) -> {
             UserModel user1 = session.users().getUserByUsername(realm, "user1");
             user1.setSingleAttribute("LDAP_ID", "WRONG");
-            return user1;
+            return user1.getId();
         });
 
-        // validate imported user
+        // validate imported user, user will be deleted and re-created
         withRealm(realmId, (session, realm) -> {
-            assertThat(session.users().getUserByUsername(realm, "user1"), is(nullValue()));;
-            UserModel deletedUser = UserStoragePrivateUtil.userLocalStorage(session).getUserByUsername(realm, "user1");
-            assertThat(deletedUser, is(nullValue()));
+            assertThat(session.users().getUserByUsername(realm, "user1"), not(equalTo(oldUserId)));
```

Comment:
```suggestion
            assertThat(session.users().getUserByUsername(realm, "user1").getId(), not(equalTo(oldUserId)));
```

This assertion should fail if the user storage provider is not implementing the `ImportedUserValidation`.

Reply:
Yes, but for this test we know it implements it, and we know that it will invalidate the existing user (by changing the ID), and we want to see that it is re-created.